manta-accounting/src/transfer/mod.rs

```diff
@@ -1106,10 +1103,11 @@ where
         I: Iterator<Item = (Self::AccountId, C::AssetValue)>;
 
     /// Checks that the transfer proof stored in `posting_key` is valid.
+    #[allow(clippy::type_complexity)] // NOTE: Clippy is too harsh here.
     fn is_valid(
         &self,
         posting_key: TransferPostingKeyRef<C, Self>,
-    ) -> Option<(Self::ValidProof, Self::Event)>;
+    ) -> Result<Option<(Self::ValidProof, Self::Event)>, <Self as TransferLedger<C>>::Error>;
```

Comment:
I kinda agree with @ghzlatarev that this would make more sense to just be a `Result<Proof,Err>` without the intermediate option, squelching clippy here is fixing the issue at the wrong end :)